map_analysis_2d/workers/peak_fitting_worker.py

```
@@ -0,0 +1,99 @@
+                            bounds=bounds,
+                            maxfev=1000,
+                        )
+
+                        ss_res = np.sum((y_fit - model_func(x_fit, *popt)) ** 2)
+                        ss_tot = np.sum((y_fit - np.mean(y_fit)) ** 2)
+                        r_squared = 1 - (ss_res / ss_tot) if ss_tot > 0 else 0.0
+
+                        for param_index, name in enumerate(param_names):
+                            results['map_parameters'][name][pos_key] = popt[param_index]
+                        results['r_squared'][pos_key] = r_squared
+                    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)